projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -23,7 +35,14 @@ export class CartesianLegend {
     private readonly injector: Injector,
     private readonly intervalData?: CartesianIntervalData,
     private readonly summaries: Summary[] = []
-  ) {}
+  ) {
+    this.activeSeries = [...this.series];
+    this.initialSeries = [...this.series];
+    this.activeSeries$ = this.activeSeriesSubject.asObservable().pipe(
```

Comment:
we shouldn't need observables for this. This could be simplified.

Reply:
Any thoughts?

Reply:
Why don't we introduce a 'visible' property in the series object (default = true). With every change, we can set visible to false. So, Active series would be a filter on series with visible = true. Would this avoid the subs and observable?

Reply:
Observables make sense to me here. We don't want to mutate the visibility of the series object that's passed in, and we'd still have to notify the chart when this changed in order to rerender.